### ISSUE TITLE
Can not reproduce DilatedNet result using provided solver, data_layer and training network 

### ISSUE THREAD

**Issue by Hexiang-Hu** [CSAILVision/sceneparsing#7]
I tried to reproduce the result with provided data layer, training network and solver parameters but fail to produce even close result as provided DilatedNet model [http://sceneparsing.csail.mit.edu/model/DilatedNet_iter_120000.caffemodel]. 

A test run on validation images with model of 120000 iteration gives me following stats: 
  - iteration 120000 overall accuracy 0.71458910259
  - iteration 120000 mean accuracy 0.321233999994
  - iteration 120000 mean IU 0.243954227299
  - iteration 120000 fwavacc 0.567641521075

However, the reported baseline performance is (73.6, 44.6, 32.3, 60.1)

I wonder what is going wrong and what I should do to have a matching result? 
[The training images are resized to 384x384 and mirrored to match the setting of author's]

**Comment by hangzhaomit** [comment]
@Hexiang-Hu Hi Hexiang, there are several tricks we used to improve the performance, like data augmentation, using pretrained model on imageNet, large mini-batch training on multiple GPUs, early stopping, etc.

**Comment by Hexiang-Hu** [comment]
@hangzhaomit Thanks for the reply. I used imageNet pretrained model as well, but only have mirror as a data augmentation method. It would be so great if you provide some details about those tricks, like how large the mini-batch is, what early stopping strategy you used and which data augmentations are adapted in this model? 

Sorry to bother you so much but I really want to reproduce the performance and do fair comparison to some other methods.

**Comment by hangzhaomit** [comment]
@Hexiang-Hu We used mini-batch size of 16; images randomly shifted to have 100k images in training. In your case, I expect you can get much better performance if you simply train for more iterations.

**Comment by Hexiang-Hu** [comment]
@hangzhaomit Thanks for the information. I will try more iterations.

**Comment by Hexiang-Hu** [comment]
I trained the network for 250k iterations and now the performance looks much nicer. Now pixel acc is around 73.2 and mean iu is 30.3. 

@hangzhaomit Do you think more training iterations will continue to improve mean IU?

**Comment by hangzhaomit** [comment]
@Hexiang-Hu Good to know. Yes, should have space for improvement.

**Comment by Hexiang-Hu** [comment]
Thank you so much

**Comment by himsR** [comment]
@Hexiang-Hu  Did you try to train the net from scratch without using the pretrained weights?I tried training the net from scratch and  I couldn't even get 70 percent accuracy. I get blank images after training using the code. Were there any tricks that you used while training your net ? I tried to initialise the net with pretrained vgg-16 weights but that did not help either. Feel like there is some trick that I am missing. Any leads appreciated. Thanks!
